internal/github/pr.go

```diff
@@ -148,3 +164,45 @@ func PRHasLabel(pr *github.PullRequest, label string) bool {
 
 	return false
 }
+
+func (ph *PRHelperImpl) GetUser(ctx context.Context, commit *object.Commit) (*github.User, error) {
+
+	email := commit.Author.Email
+	userSearchRes, _, err := ph.gc.Search.Users(ctx, email, nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not get user from email %s: %v", email, err)
+	}
+	if len(userSearchRes.Users) != 1 {
+		return nil, fmt.Errorf("there is more than 1 user associated with %s: %v", email, err)
+	}
+	user := userSearchRes.Users[0]
+
+	dsOwner := ph.repoName.Owner
+	dsRepo := ph.repoName.Repo
+	filePath := "./OWNERS"
+	content, _, _, err := ph.gc.Repositories.GetContents(ctx, dsOwner, dsRepo, filePath, nil)
+	if err != nil || content != nil { // `content` can be nil if the filePath refers to a directory
+		return nil, fmt.Errorf("could not get %s's content from %s/%s: %v", filePath, dsOwner, dsRepo, err)
+	}
+	data, err := content.GetContent()
+	if err != nil {
+		return nil, fmt.Errorf("%s's data is invalid: %v", filePath, err)
+	}
+
+	if strings.Contains(data, *user.Login) {
```

Comment:
I think we need something more robust here. I would parse `OWNERS` as a YAML file and build a set of usernames for faster lookups.

Reply:
Fixed.